pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -351,3 +350,52 @@ def on_train_batch_start(self, batch: Any, batch_idx: int, dataloader_idx: int)
         Called in the training loop before anything happens for that batch.
         """
         pass
+
+    def pre_dispatch(self) -> None:
+        """Hook to do something before the training/evaluation/prediction starts."""
+
+    def dispatch(self, trainer: "pl.Trainer") -> None:
+        """Hook to do something at trainer run_stage starts."""
+
+    def post_dispatch(self) -> None:
+        """Hook to do something after the training/evaluation/prediction finishes."""
+
+    @contextlib.contextmanager
+    def train_step_context(self) -> Generator:
+        """
+        A contextmanager for the trainstep
+
+        .. deprecated::v1.5
+            Will be removed in v1.7.0.
+        """
+        yield
+
+    @contextlib.contextmanager
+    def val_step_context(self) -> Generator:
+        """
+        A contextmanager for the validation step
+
+        .. deprecated::v1.5
+            Will be removed in v1.7.0.
```

Comment:
I vote for skipping deprecation and go with direct removal

Reply:
+1

Reply:
There is reference to training_type_plugin.train_step_context, training_type_plugin.val_step_context, training_type_plugin.test_step_context and training_type_plugin.predict_step_context in accelerator. example: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/accelerators/accelerator.py#L222
Non of the lightning training type plugin overrides these functions, but we have no way to confirm that custom plugin overrides is not using it? I think probably deprecate is safer for now

Reply:
> Non of the lightning training type plugin overrides these functions, but we have no way to confirm that custom plugin overrides is not using it? I think probably deprecate is safer for now

From the docs: The Plugin API is in beta and subject to change. For help setting up custom plugins/accelerators, please reach out to us at support@pytorchlightning.ai

https://pytorch-lightning.readthedocs.io/en/latest/extensions/plugins.html

Given this is experimental, I think it is better for us to do an outright removal instead of going through the full deprecation process: https://pytorch-lightning.readthedocs.io/en/latest/governance.html#api-evolution

It is a breaking change; but as no plugin supported directly by lightning is affected, i think this is reasonable. if this broke something like DDP, that would not be okay

Reply:
make sense, removed!